powertools-serialization/src/main/java/software/amazon/lambda/powertools/utilities/EventDeserializer.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package software.amazon.lambda.powertools.utilities;
+
+import com.amazonaws.services.lambda.runtime.events.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static software.amazon.lambda.powertools.utilities.jmespath.Base64Function.decode;
+import static software.amazon.lambda.powertools.utilities.jmespath.Base64GZipFunction.decompress;
+
+public class EventDeserializer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EventDeserializer.class);
+
+    public static EventPart extractDataFrom(Object obj) {
+        if (obj instanceof String) {
+            return new EventPart((String) obj);
+        } else if (obj instanceof Map) {
+            return new EventPart((Map<String, Object>) obj);
+        } else if (obj instanceof APIGatewayProxyRequestEvent) {
+            APIGatewayProxyRequestEvent event = (APIGatewayProxyRequestEvent) obj;
+            return new EventPart(event.getBody());
+        } else if (obj instanceof APIGatewayV2HTTPEvent) {
+            APIGatewayV2HTTPEvent event = (APIGatewayV2HTTPEvent) obj;
+            return new EventPart(event.getBody());
+        } else if (obj instanceof SNSEvent) {
+            SNSEvent event = (SNSEvent) obj;
+            return new EventPart(event.getRecords().get(0).getSNS().getMessage());
+        } else if (obj instanceof SQSEvent) {
+            SQSEvent event = (SQSEvent) obj;
+            return new EventPart(event.getRecords().stream().map(SQSEvent.SQSMessage::getBody).collect(Collectors.toList()));
```

Comment:
Please indent streaming in all if else branches, where applicable to have one operation per line. I know we have limited options to make if else prettier but atleast properly indenting things, make it much more readable.  for example

```suggestion
return new EventPart(event.getRecords().stream()
                    .map(SQSEvent.SQSMessage::getBody)
                    .collect(toList()));
```

Reply:
OK